tests/unit/hooks/test_pushover.py

```diff
@@ -0,0 +1,537 @@
+    as an 'advanced example'. This test runs the finish state.
+    '''
+    hook_config = {
+        'token': 'ksdjfwoweijfvwoeifvjmwghagy92',
+        'user': '983hfe0of902lkjfa2amanfgui',
+        'states': {'start', 'fail', 'finish'},
+        'start': {
+            'message': 'Backup <b>Started</b>',
+            'priority': -2,
+            'title': 'Backup Started',
+            'html': 1,
+            'ttl': 10,
+        },
+        'fail': {
+            'message': 'Backup <font color="#ff6961">Failed</font>',
+            'priority': 2,
+            'expire': 600,
+            'retry': 30,
+            'device': 'pixel8',
+            'title': 'Backup Failed',
+            'html': 1,
+            'sound': 'siren',
+            'url': 'https://ticketing-system.example.com/login',
+            'url_title': 'Login to ticketing system',
+        },
+        'finish': {
+            'message': 'Backup <font color="#77dd77">Finished</font>',
+            'priority': 0,
+            'title': 'Backup Finished',
+            'html': 1,
+            'ttl': 60,
+            'url': 'https://ticketing-system.example.com/login',
+            'url_title': 'Login to ticketing system',
+        },
+    }
+    flexmock(module.logger).should_receive('warning').never()
+    flexmock(module.requests).should_receive('post').with_args(
+        'https://api.pushover.net/1/messages.json',
+        headers={'Content-type': 'application/x-www-form-urlencoded'},
+        data={
+            'token': 'ksdjfwoweijfvwoeifvjmwghagy92',
+            'user': '983hfe0of902lkjfa2amanfgui',
+            'message': 'Backup <font color="#77dd77">Finished</font>',
+            'priority': 0,
+            'title': 'Backup Finished',
+            'html': 1,
+            'ttl': 60,
+            'url': 'https://ticketing-system.example.com/login',
+            'url_title': 'Login to ticketing system',
+        },
+    ).and_return(flexmock(ok=True)).once()
+
+    module.ping_monitor(
+        hook_config,
+        {},
+        'config.yaml',
+        borgmatic.hooks.monitor.State.FINISH,
+        monitoring_log_level=1,
+        dry_run=False,
+    )
+
+
+def test_ping_monitor_config_with_minimum_config_fail_state_backup_successfully_send_to_pushover_dryrun():
+    '''
+    This test should be the minimum working configuration. The "message"
+    should be auto populated with the default value which is the state name.
+    '''
+    hook_config = {'token': 'ksdjfwoweijfvwoeifvjmwghagy92', 'user': '983hfe0of902lkjfa2amanfgui'}
+    flexmock(module.logger).should_receive('warning').never()
+    flexmock(module.requests).should_receive('post').with_args(
+        'https://api.pushover.net/1/messages.json',
+        headers={'Content-type': 'application/x-www-form-urlencoded'},
+        data={
+            'token': 'ksdjfwoweijfvwoeifvjmwghagy92',
+            'user': '983hfe0of902lkjfa2amanfgui',
+            'message': 'fail',
+        },
```

Comment:
I don't think you need `.with_args(...)` here, because you basically want an expectation that `post` is never called—regardless of the arguments it receives.